qiime2/sdk/util.py

```diff
@@ -183,3 +185,26 @@ def get_available_usage_drivers() -> Dict[str, 'UsageDriver']:
         entry_point.name: entry_point.resolve() for entry_point in
         iter_entry_points(group='qiime2.usage_drivers')
     }
+
+
+def get_available_cores(one_less: bool = False):
```

Comment:
Put this in `qiime2.plugin` if it's for plugins, and I would actually recommend `psutil` it's a bit more robust than the stdlib for this things (and we already depend on it).